### PR TITLE
Use ninja for build

### PR DESCRIPTION
**Pull request by dg0yt**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by dg0yt** [comment]
CI build time changes:

| Target | [Before this PR](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=829474&view=results) | [`ninja`, this PR](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=830379&view=results) | Remark |
|----------|--------------------|---------------------|-------------|
| linux          | 30 min | 42 min | 12 min worse - unexpected! |
| win            | 55 min | 43 min | 12 min improvement |
| osx x64     | 54 min | 45 min | 9 min improvement |
| osx arm64 | 50 min | 42 min | 8 min improvement |

NB: Build times in CI vary widely, but I only took single data points.

## linux-64
`make` before this PR:
~~~
Resource usage statistics from building scipoptsuite:
   Process count: 16
   CPU time: Sys=0:00:57.0, User=0:27:37.2
   Memory: 5.0G
   Disk usage: 381.1K
   Time elapsed: 0:20:18.6
~~~

`ninja` with this PR:
~~~
Resource usage statistics from building scipoptsuite:
   Process count: 11
   CPU time: Sys=0:01:32.3, User=0:39:43.2
   Memory: 5.0G
   Disk usage: 318.0K
   Time elapsed: 0:27:36.9
~~~

I did expect at least a small improvement, similar to osx. Something strange going on here. Or simply a machine with less cores (cf. Process count).

**Comment by dg0yt** [comment]
`ninja` on another CI run:
~~~
Resource usage statistics from building scipoptsuite:
   Process count: 11
   CPU time: Sys=0:00:56.0, User=0:27:18.5
   Memory: 5.0G
   Disk usage: 154.6M
   Time elapsed: 0:19:43.0
~~~
So everything is okay, only CI machines vary.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -28,6 +28,7 @@ requirements:
     - {{ compiler('cxx') }}
     - cmake >=3.15
     - make            # [unix]
```
Comment (h-vetinari) [inline recipe/meta.yaml]: You can remove make here if you're switching to ninja.
```suggestion
```

**Comment by h-vetinari** [comment]
> So everything is okay, only CI machines vary.

Yeah, the machines we get randomly assigned in Azure Pipelines can be quite different in capabilities. Depending on the recipe, a factor 2x can happen between the fastest and slowest build times.

File: recipe/meta.yaml (hunk 1 above)
Comment (dg0yt) [reply recipe/meta.yaml]: I left it because of the examples. But we can switch them to ninja, too.

**Review by AntoinePrv — APPROVED** [review]

**Pull request merged**